File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/gamepad/GamepadDataProvider.java
package net.kdt.pojavlaunch.customcontrols.gamepad;

import net.kdt.pojavlaunch.GrabListener;

public interface GamepadDataProvider {
/**
 * 「MenuMap」の値を取得します。
 */
    GamepadMap getMenuMap();
/**
 * 「GameMap」の値を取得します。
 */
    GamepadMap getGameMap();
/**
 * このオブジェクトが「Grabbing」状態であるかを判定します。
 */
    boolean isGrabbing();
/**
 * 「attach Grab Listener」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    void attachGrabListener(GrabListener grabListener);
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/gamepad/GamepadDataProvider.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.gamepad

import com.arata.yukarilauncher.ui.view.GrabListener

/** ゲームパッドのマップデータを提供するインターフェース。 */
interface GamepadDataProvider {
    /** メニュー画面用のゲームパッドマップ。 */
    val menuMap: GamepadMap
    /** ゲーム内用のゲームパッドマップ。 */
    val gameMap: GamepadMap
    /** @return 現在グラブ中かどうか */
    fun isGrabbing(): Boolean
    /** @param grabListener グラブリスナーをアタッチします */
    fun attachGrabListener(grabListener: GrabListener)
}